factor/combinatoric.py

```diff
@@ -1,9 +1,20 @@
 
 
 def factorial(n: int) -> int:
-    # TODO implement for +10 bonus
-    return 0
+    # If n is 1, return 1
+    if n == 1:
+        return 1
+    # Otherwise, return n times factorial of (n - 1)
+    return n * factorial(n - 1)
 
 def choose(n: int, k: int) -> int:
-    # TODO implement for +10 bonus
-    return 0
\ No newline at end of file
+    # If k is more than half of n, use n - k instead
+    if k > n - k:
+        k = n - k
+    # Start with result as 1
+    result = 1
+    # Loop to calculate the combination
+    for i in range(1, k + 1):
```

Comment:
this looks correct, but why didn't you leverage the factorial function you had already written? additionally, we are missing the test cases in test_combinatoric.py